docs/docs/developers/contracts/references/storage/shared_state.md

```diff
@@ -0,0 +1,137 @@
+
+This function can only be called in public, typically after some access control check:
+
+#include_code shared_mutable_schedule /noir-projects/noir-contracts/contracts/auth_contract/src/main.nr rust
+
+If one wishes to schedule a value change from private, simply enqueue a public call to a public `internal` contract function. Recall that **all scheduled value changes, including the new value and scheduled block are public**.
+
+:::warning
+A `SharedMutable`'s storage **must** only be mutated via `schedule_value_change`. Attempting to override this by manually accessing the underlying storage slots breaks all properties of the data structure, rendering it useless.
+:::
+
+### `get_current_value_in_public`
+
+Returns the current value in a public execution context. Once a value change is scheduled via `schedule_value_change` and a number of blocks equal to the delay passes, this automatically returns the new value.
+
+#include_code shared_mutable_get_current_public /noir-projects/noir-contracts/contracts/auth_contract/src/main.nr rust
+
+### `get_current_value_in_private`
+
+Returns the current value in a private execution context. Once a value change is scheduled via `schedule_value_change` and a number of blocks equal to the delay passes, this automatically returns the new value.
+
+Calling this function will set the `max_block_number` property of the transaction request, introducing a new validity condition to the entire transaction: it cannot be included in any block with a block number larger than `max_block_number`. This could [potentially leak some privacy](#privacy-considerations).
+
+#include_code shared_mutable_get_current_private /noir-projects/noir-contracts/contracts/auth_contract/src/main.nr rust
+
+### `get_scheduled_value_in_public`
+
+Returns the last scheduled value change, along with the block number at which the scheduled value becomes the current value. This may either be a pending change, if the block number is in the future, or the last executed scheduled change if the block number is in the past (in which case there are no pending changes).
+
+#include_code shared_mutable_get_scheduled_public /noir-projects/noir-contracts/contracts/auth_contract/src/main.nr rust
+
+It is not possible to call this function in private: doing so would not be very useful at it cannot be asserted that a scheduled value change will not be immediately replaced if `shcedule_value_change` where to be called.
+
+## `SharedImmutable`
```

Comment:
The shared immutable is a bit weird for me to have down here as it is not really using much of the things that was just walked through. It is a lot easier to understand so might make sense to have at first

Reply:
I went back and forth on this. Yes it's simpler to understand, but it's also not terribly interesting. I definitely think it fits better in the shared state page than in the public one (especially after adding a link here from there), but I'd rather highlight that shared state is hard and has limitations instead of starting by introducing a glorified constant.